Sources/ApolloCodegenLib/Templates/CocoaPodsModuleTemplate.swift

```diff
@@ -0,0 +1,24 @@
+import Foundation
+
+struct CocoaPodsModuleTemplate {
+  let name: String
+  let version: String
+  let license: String
+  let homepage: URL
+  let source: URL
+
+  func render() -> String {
+    TemplateString("""
```

Comment:
These are the [minimum required fields for a podspec](https://guides.cocoapods.org/syntax/podspec.html#group_root_specification), some of them we can autogenerate the content but most are really quite specific to the developer/organization/project and not something we can glean from the codegen configuration.

Right now these are stored as associated values with [the `.cocoapods` enum case of ModuleType](https://github.com/apollographql/apollo-ios/pull/2185/files#diff-c75e72335519c23675c4ac6e74cd5913ed6265a4450bdc32915044f6b65680c0R162). That means a [larger configuration](https://github.com/apollographql/apollo-ios/pull/2185/files#diff-b3d18e19585b1dcca04806cc30099bebe006b1f067427fb9eebda32ed083af29R6) but does provide a lot of flexibility to the user. The podspec specification seems to be quite stable so I'm confident this isn't something we'll have to change often, if at all.

An alternative is to simply have a single freeform `String` that the user must specify that is the 'guts' of the podspec but that opens a whole bunch of potential error scenarios where users have to consult the podspec documentation and then format it correctly - it's just not a good idea.

@AnthonyMDev would love your thoughts on this and whether you have alternative suggestions.

Reply:
This also isn't complete because the `source_files` field has a bunch of variability to it which will need more associated values.